Feature/data processing module

vinayaknatarajan opened the pull request:
- Module to fetch Inspection data from NYC OPEN Data through Socrata API.
- Clean and preprocess the data
- Module to save data into Restaurant Table
- Module to save data into the Inspection table.

vinayaknatarajan committed:
- get Inspection Data
- Clean inspection data
- Create restaurants and inspection datframes

vinayaknatarajan committed:
- Save restaurant records
- Save inspectionRecords

vinayaknatarajan committed:
Delete DineSafely.iml

vinayaknatarajan committed:
Delete profiles_settings.xml

vinayaknatarajan committed:
Delete modules.xml

vinayaknatarajan committed:
Delete misc.xml

vinayaknatarajan committed:
Delete vcs.xml

chenwei713 commented on restaurant/models.py:
```diff
@@ -5,25 +5,22 @@ class Restaurant(models.Model):
     postcode = models.CharField(max_length=200)
-    business_id = models.CharField(max_length=200, default=None)
+    business_id = models.CharField(max_length=200, default=None,blank=True,null = True)
+    legal_business_name = models.CharField(max_length=200,default = None)
```
I don't think we need legal_business_name here.
For example, different restaurants belong to the same big company, they should have the same legal business name. And the legal business name doesn't help when we query the Yelp.

deusalexmachina replied:
I agree, unless there is good reason why we include this. restaurant name, address and zip code should be good enough for now.

chenwei713 commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+
```
files in .idea and .pyc files should not be included

chenwei713 commented on getinspection.py:
```diff
@@ -0,0 +1,62 @@
+
```
I think it's better to remove the blank lines and useless comments.
I suggest changing the name of this file and moving this python script to a folder called scripts instead of at the root of the application.

Is this script only for the first time we populate the database? Can we reuse it later?

deusalexmachina replied:
maybe we can make the code more modular so that we can run a script for initial population and another for scheduler updates?

vinayaknatarajan replied:
As of now, we run it only once. But we will be adding a scheduler on Heroku in the next sprint which will call this function to populate the database at regular intervals.

chenwei713 commented on restaurant/models.py:
```diff
@@ -5,25 +5,22 @@ class Restaurant(models.Model):
     business_address = models.CharField(max_length=200)
     postcode = models.CharField(max_length=200)
-    business_id = models.CharField(max_length=200, default=None)
+    business_id = models.CharField(max_length=200, default=None,blank=True,null = True)
```
null=True

deusalexmachina commented on getinspection.py:
```diff
@@ -0,0 +1,51 @@
+    print(restaurant_df.shape)
+    return restaurant_df, inspection_df
+
+# def saveRestaurants(restaurant_df):
```
I think we should remove the commented out lines.

deusalexmachina commented on restaurant/models.py:
```diff
@@ -6,24 +6,20 @@ class Restaurant(models.Model):
-    seating_choice = models.CharField(max_length=200)
-    legal_business_name = models.CharField(max_length=200)
-    business_address = models.CharField(max_length=200)
     is_sideway_compliant = models.CharField(max_length=200)
```
i suggest removing this field from the model since we don't use this column, and we won't have to collect this information from the dataset.

vinayaknatarajan committed:
- Resolved code Review comments
- changed unique constraint in Restaurant model
- Added Exception handling to Save modules
-

vinayaknatarajan committed:
Merge branch 'feature/data_processing_module' of https://github.com/gcivil-nyu-org/dine-safe-ly into feature/data_processing_module

Chuanqi927 commented:
make sure all pyc files are deleted before pushing XD

vinayaknatarajan committed:
removed *.pyc files
